charmhelpers/core/hookenv.py

```diff
@@ -964,7 +966,16 @@ def charm_dir():
     return os.environ.get('CHARM_DIR')
 
 
+def cmd_exists(cmd):
+    """Return True if the specified cmd exists in the path"""
+    return any(
+        os.access(os.path.join(path, cmd), os.X_OK)
+        for path in os.environ["PATH"].split(os.pathsep)
+    )
+
+
 @cached
+@deprecate("moved to function_get()", "2019-11", log=log)
```

Comment:
I didn't even realize we had this decorator.  I like it, but I think it's more important for the deprecation message to go into the docs.  I'd recommend using the syntax used by [`service_name`](https://github.com/juju/charm-helpers/blob/master/charmhelpers/core/hookenv.py#L237).  It would be nice if the decorator was automatically turned into the doc addition as well, but that's for future work.

Also, the date is supposed to be when the function will actually be removed from the code base, not when it is deprecated, though I don't know if anything will ever actually be removed, so it's probably fine as is.

Reply:
I removed the date arg and added "deprecated" to the doc strings